src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -356,18 +356,76 @@ protected virtual void AddRecipients(TransactionBuildContext context)
             if (context.Recipients.Any(a => a.Amount == Money.Zero))
                 throw new WalletException("No amount specified.");
 
-            if (context.Recipients.Any(a => a.SubtractFeeFromAmount))
-                throw new NotImplementedException("Substracting the fee from the recipient is not supported yet.");
+            var totalRecipients = context.Recipients.Count(r => r.SubtractFeeFromAmount);
 
-            foreach (Recipient recipient in context.Recipients)
-                context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount);
+            // If we have any recipients that require a fee to be subtracted from the amount, then
+            // calculate fee and evenly distribute it among all recipients. Any remaining fee should be
+            // subtracted from the first recipient.
+            if (totalRecipients > 0)
+            {
```

Comment:
At the risk of complicating the code, can this logic not go in to the method AddFee and in that method if falling in the condition `if (context.Recipients.Any(r => r.SubtractFeeFromAmount))` then apply this logic to the recipients already in the builder?

Reply:
@dangershony I thought about doing it, but the issue is that I would have to replace send builder then, as transaction send builder is created with UTXO in AddRecipients

Reply:
No that's even more complicated I agree, I thought there would be the ability to have direct access to the outputs, like how @MithrilMan did when adding the secrets, maybe not.